eclass/java-pkg-2.eclass

```diff
@@ -40,9 +41,15 @@ DEPEND="${JAVA_PKG_E_DEPEND}"
 RDEPEND="${DEPEND}"
 
 # Commons packages follow the same rules so do it here
-if [[ ${CATEGORY} = dev-java && ${PN} = commons-* ]]; then
-	HOMEPAGE="http://commons.apache.org/${PN#commons-}/"
-	SRC_URI="mirror://apache/${PN/-///}/source/${P}-src.tar.gz"
+if [[ ${EAPI} != 6 ]]; then
+	if [[ ${CATEGORY} = dev-java && ${PN} = commons-* ]]; then
+		HOMEPAGE="https://commons.apache.org/${PN#commons-}/"
+		SRC_URI="mirror://apache/${PN/-///}/source/${P}-src.tar.gz
+			verify-sig? ( https://downloads.apache.org/${PN/-///}/source/${P}-src.tar.gz.asc )"
+	
+		BDEPEND+="verify-sig? ( sec-keys/openpgp-keys-apache-commons )"
+		VERIFY_SIG_OPENPGP_KEY_PATH="/usr/share/openpgp-keys/commons.apache.org.asc"
+	fi
```

Comment:
i don't really like it here. it would be imo better, if we wanna do it this way and save the coding, to put it into a separate eclass and inherit that eclass when needed.

Reply:
> i don't really like it here. it would be imo better, if we wanna do it this way and save the coding, to put it into a separate eclass and inherit that eclass when needed.

Moving to a separate eclass would also allow to cover packages like `bcel` and `bsf` where `SRC_URI` wouldn't match with the present solution. Will try that within next days.